c++/src/MemoryPool.cc

```diff
@@ -86,10 +86,6 @@ namespace orc {
       for (uint64_t i = currentSize_; i > newSize; --i) {
         (buf_ + i - 1)->~T();
       }
-    } else if (newSize > currentSize_) {
-      for (uint64_t i = currentSize_; i < newSize; ++i) {
-        new (buf_ + i) T();
```

Comment:
I don't think we can remove these lines.

Reply:
There is performance benefits if we remove it. In fact, similar optimization exists in ClickHouse.  
If you find any code relying on it, please let me know. At least removing it doesn't influence the CI.

Reply:
reverted